Yard powered docs

Zeragamba opened the pull request:
Closes https://github.com/stympy/faker/issues/1330

SpyMaster356 commented:
forgot to actually open this PR for #1330

richardbulger commented:
Can CONTRIBUTING.md be updated with guidelines on how this should be used going forward?

Zeragamba commented:
Updated contributing, and added some sample documentation to [Faker::Alphanumeric], [Faker::Book], and [Faker::Books]'s subclasses.

[Faker::Alphanumeric]: https://www.rubydoc.info/github/zeragamba/faker/70db4c/Faker/Alphanumeric
[Faker::Book]: https://www.rubydoc.info/github/zeragamba/faker/70db4c/Faker/Book
[Faker::Books]: https://www.rubydoc.info/github/zeragamba/faker/70db4c/Faker/Books

connorshea commented:
👍 this'd be great, especially for use with [sord](https://github.com/AaronC81/sord) so it'd be easier to add Sorbet support (either officially in this repo or via sorbet-typed)

vbrazo commented:
@Zeragamba can you update this branch with master? We'll move on with these changes for v3.

Zeragamba commented:
Sure!

On Wed, Aug 7, 2019, 00:09 Vitor Oliveira, <notifications@github.com<mailto:notifications@github.com>> wrote:

@Zeragamba<https://github.com/Zeragamba> can you update this branch with master? We'll move on with these changes for v3.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/faker-ruby/faker/pull/1553?email_source=notifications&email_token=AARXXYM5VGQYDVK27NDNM4TQDJDF7A5CNFSM4GZY7C22YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD3XEIRY#issuecomment-518931527>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AARXXYN56FSROSLKVICKUD3QDJDF7ANCNFSM4GZY7C2Q>.

bpleslie commented on CONTRIBUTING.md:
```diff
@@ -43,32 +43,64 @@ If you're reviewing a PR, you should ask yourself:
-  
-# instead of these: 
+* In general, follow the conventions you see used in the source already.
+* **ALL SHALL OBEY THE RUBOCOP**
```
👏

Zeragamba commented on lib/faker/blockchain/tezos.rb:
```diff
@@ -17,28 +18,78 @@ class << self
         end
 
+        ##
+        # Produces a random Tezos block
```
@akettal Is there a better summary I could put in here?

connorshea commented on lib/faker/creature/dog.rb:
```diff
@@ -6,34 +6,106 @@ class Dog < Base
+        # @return [String]
+        #
+        # @example Faker::Creature::Dog.gender
+        #   "female"
```
This should be `Female`, capitalized :)

```suggestion
        #   "Female"
```

connorshea commented on lib/faker/default/alphanumeric.rb:
```diff
@@ -3,16 +3,45 @@
+      ##
+      # Produces a random string of alphanumeric characters
+      #
+      # @param char_count [Integer] The length of the string to generate
```
```suggestion
      # @param number [Integer] The length of the string to generate
```

connorshea commented on lib/faker/default/alphanumeric.rb:
```diff
@@ -3,16 +3,45 @@
+      #
+      # @return [String]
+      #
+      # @example Faker::Alphanumeric.alpha(10)
```
```suggestion
      # @example Faker::Alphanumeric.alpha(number: 10)
```

connorshea commented:
Also, you should probably add `doc/` to the `.gitignore` since running `yard` causes a bunch of files to be added.

connorshea commented:
For the `@example`s, you're formatting them like this:

```ruby
# @example Faker::Game.genre
#   "Real-time strategy"
```

when normally it would be more like:

```ruby
# @example 
#   Faker::Game.platform #=> "Nintendo Switch"
```

You can see the rendered documentation for these two here:

<img width="1135" alt="Screen Shot 2019-08-12 at 8 05 24 PM" src="https://user-images.githubusercontent.com/2977353/62910114-8c3ce800-bd3c-11e9-82ef-0a6d03916255.png">

I can make this change if you want it.

stympy commented:
I do prefer the latter style, @connorshea

connorshea commented:
I think I've updated all of them now, that wasn't as hard as I expected. https://github.com/Zeragamba/faker/pull/6

connorshea commented:
@vbrazo just to clarify, you’re not saying this PR would need to wait until 3.0, right?

We can have the current docs and the YARD docs simultaneously without any problem, and the current docs would be the main source until we’ve moved everything over to YARD.

Zeragamba commented:
@connorshea Sounds good. I saw your pull request over 'ere: https://github.com/Zeragamba/faker/pull/6 (and linking for others following this thread)

Left a few change requests there, I'll merge those in when one of us get the time to fix it. :)

connorshea commented on .gitignore:
```diff
@@ -23,3 +23,5 @@ coverage/*
 .ruby-version
 .ruby-gemset
+
+.yardoc/
```
To prevent the docs generated by `yard` from being committed :)

```suggestion
.yardoc/
doc/
```

connorshea replied:
Oh I just realized the `doc/` directory is already used. We should probably set a custom output directory in the yard config.

Zeragamba commented:
Lots of new docs from @connorshea so this is ready for re-review

connorshea commented:
There are some new conflicts now.

Zeragamba commented on lib/faker/blockchain/bitcoin.rb:
```diff
@@ -7,21 +7,49 @@ module Faker
+        #
+        # @return [String]
+        #
+        # @return [String]
```
double returns whoops

Zeragamba commented on CONTRIBUTING.md:
```diff
@@ -1,74 +1,109 @@
+### Syntax/Good practices:
+
+#### Documentation
+Include [YARD] style docs for all methods that includes:
```
should also mention which tags to use here too

Zeragamba commented on lib/faker/books/lovecraft.rb:
```diff
@@ -16,14 +37,46 @@ def fhtagn(legacy_number = NOT_GIVEN, number: 1)
+        # @return [String]
+        #
+        # @example
+        #   Faker::Books::Lovecraft.sentence #=> "Furtive antiquarian squamous dank cat loathsome amorphous lurk."
```
need to split into more lines

Zeragamba commented on lib/faker/books/lovecraft.rb:
```diff
@@ -75,6 +207,28 @@ def paragraph(legacy_sentence_count = NOT_GIVEN, legacy_random_sentences_to_add
+        # @example
+        #   Faker::Books::Lovecraft.paragraphs
+        #   #=> [
+        #   #     "Noisome daemoniac gibbous abnormal antediluvian. Unutterable fungus accursed stench noisome lurk madness indescribable. Antiquarian fungus gibbering lurk dank fainted. Hideous loathsome manuscript daemoniac lurk charnel foetid.",
```
probably should trim these lines

Zeragamba commented:
live example of these docs: https://rubydoc.info/github/SpyMaster356/faker/master/Faker/Avatar

Zeragamba commented:
It merged! Woooo :party:

connorshea commented:
🎉
